packages/components/src/components/avatar/avatar.stories.ts

```diff
@@ -1,29 +1,133 @@
 import type { Meta, StoryObj, Args } from '@storybook/web-components';
+import { repeat } from 'lit/directives/repeat.js';
 import '.';
 import { html } from 'lit';
+import { TYPE as PRESENCE_TYPE, SIZE as PRESENCE_SIZE } from '../presence/presence.constants';
 import { DEFAULTS } from './avatar.constants';
 
 const render = (args: Args) => html`
-  <mdc-avatar src="${args.src}" size="${args.size}" type="${args.type}"></mdc-avatar>
+  <mdc-avatar
+    alt="${args.alt}"
+    aria-label="${args['aria-label']}"
+    counter="${args.counter}"
+    icon-name="${args['icon-name']}"
+    initials="${args.initials}"
+    presence="${args.presence}"
+    size="${args.size}"
+    src="${args.src}"
+    ?is-clickable="${args['is-clickable']}"
+    ?is-typing="${args['is-typing']}"
+  ></mdc-avatar>
 `;
 
 const meta: Meta = {
   title: 'Work In Progress/avatar',
+  tags: ['autodocs'],
   component: 'mdc-avatar',
   render,
   parameters: {
     badges: ['wip'],
   },
-  argTypes: {},
+  argTypes: {
+    src: {
+      control: 'text',
+    },
+    initials: {
+      control: 'text',
+    },
+    presence: {
+      if: {
+        arg: 'counter',
+        exists: false,
+      },
+      control: 'select',
+      options: Object.values(PRESENCE_TYPE),
+    },
+    size: {
+      control: 'select',
+      options: Object.values(PRESENCE_SIZE),
+    },
+    'icon-name': {
+      control: 'text',
+    },
+    counter: {
+      control: 'number',
+    },
+    'is-typing': {
+      control: 'boolean',
+    },
+    'aria-label': {
+      control: 'text',
+    },
+  },
 };
 
 export default meta;
 
-export const Primary: StoryObj = {
+export const Image: StoryObj = {
+  args: {
+    size: 'xx_large',
+    src: 'https://picsum.photos/id/63/256',
+    initials: 'AK',
+    'icon-name': 'placeholder-bold',
+    'aria-label': 'An Avatar with Photo',
+  },
+};
+
+export const Default: StoryObj = {
   args: {
-    // eslint-disable-next-line max-len
-    src: 'https://images.unsplash.com/photo-1583195764036-6dc248ac07d9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=2855&q=80',
     size: DEFAULTS.SIZE,
-    type: DEFAULTS.TYPE,
+  },
+};
+
+export const Initials: StoryObj = {
+  args: {
+    initials: 'MD',
+    size: 'xx_large',
+  },
+};
```

Comment:
```suggestion
export const Initials: StoryObj = {
 argTypes: {
    ...disableControls(['counter', 'icon-name']),
  },
  args: {
    initials: 'MD',
    size: 'xx_large',
  },
};
```

suggestion: can we restrict the attributes for specific avatar type examples. When I am testing them individual, it is causing confusion.
For example: On the 'Initial' story, we have an option to set counter. However, if I set the counter value, the avatar does not change to a counter type (it still remains as an text avatar).

Reply:
The same restriction can be applied conditionally for each story component